File: dto/src/main/java/io/github/manami/dto/entities/MinimalEntry.java
package io.github.manami.dto.entities;

import static org.apache.commons.lang3.StringUtils.isNotBlank;

/**
 * @author manami-project
 * @since 2.7.0
 */
public interface MinimalEntry {

    String getTitle();


    String getThumbnail();


    String getInfoLink();


    /**
     * Checks if a MinimalEntry entry is Valid
     * @param anime
     * @return
     */
    static boolean isValidMinimalEntry(final MinimalEntry anime) {
        boolean ret = anime != null;

        if (!ret) {
            return ret;
        }

        ret &= isNotBlank(anime.getTitle());
        ret &= isNotBlank(anime.getInfoLink());

        return ret;
    }
}

File: dto/src/main/kotlin/io/github/manami/dto/entities/MinimalEntry.kt
package io.github.manami.dto.entities

import java.net.URL


interface MinimalEntry {
    var title: String

    var thumbnail: URL

    var infoLink: InfoLink

    fun isValidMinimalEntry() = title.isNotBlank() && infoLink.isValid()

    companion object {
        /** Placeholder image in case no image is available. */
        val NO_IMG = URL("https://myanimelist.cdn-dena.com/images/na_series.gif")

        /** Placeholder image in case no image is available, thumbnail size. */
        val NO_IMG_THUMB = URL("https://myanimelist.cdn-dena.com/images/qm_50.gif")
    }
}
